File: app/src/main/java/com/rendersoncs/reportform/extension/StringExtension.java
package com.rendersoncs.reportform.extension;

public class StringExtension {

    public static String limitsText(String text, int i) {
        if (text.length() > i) {
            int startIndex = 0;
            StringBuilder stringBuilder = new StringBuilder();
            String textSize = text.substring(startIndex, i);
            return stringBuilder.append(textSize).append("...").toString();
        } else {
            return text;
        }
    }
}

File: app/src/main/java/com/rendersoncs/report/common/extension/StringExtension.kt
package com.rendersoncs.report.common.extension

object StringExtension {
    @JvmStatic
    fun limitsText(text: String?, i: Int): String {
        return if (text?.length!! > i) {
            val startIndex = 0
            val stringBuilder = StringBuilder()
            val textSize = text.substring(startIndex, i)
            stringBuilder.append(textSize).append("...").toString()
        } else {
            text
        }
    }
}

fun String.spaceToNewLine(): String {
    return try {
        val index = this.indexOf(' ')
        if (index != -1 && index < this.length - 1) {
            this.replaceFirst(" ", "\n")
        } else { this }
    } catch (exception: Exception) {
        this
    }
}